### DISCUSSION TITLE
providerId always 'firebase'

### DISCUSSION THREAD

**Discussion by Danielmn94** [capawesome-team/capacitor-firebase#363] [Q&A] [answered]
Hello.

Is it the intended way when getting user, that providerId is 'firebase' even though i'm logged in either via facebook or google?

![image](https://user-images.githubusercontent.com/26299136/232573713-293f986d-d3ca-4bdc-ab5e-27fa7f14bfed.png)

I thought the providerId should represent the providerId enum.

Best regards.

**Comment by robingenz** [comment]
Please take a look at this [comment](https://github.com/invertase/react-native-firebase/issues/1768#issuecomment-448604982).
The `providerId` you are looking for should be available here: https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/authentication#additionaluserinfo

**Reply by Danielmn94** [reply]
Hi.

Thanks for the fast reply. Appreciate it

When using FirebaseAuthentication.getCurrentUser(); i'm not getting a property called providerData or am i doing something wrong?
And can't seem to find providerData when reading the readme that you linked.

And regards to AddtionalUserInfo isn't that only when you create a user you get that object, or is there a way to get this information when a user is logged in?

My problem is i need to see if a user is logged in using facebook, google, email/password, etc. Before using capacitor i could loop over the providerData array. Or do i need to user firebase/auth and capacitor-firebase/authentication simultaneously?

**Reply by robingenz** [reply, marked as answer]
This property is currently only available at sign-in. There is an open feature request to add "providerData" to "getCurrentUser()": #161

If you want, you can use the plugin in combination with the Firebase JS SDK (see [here](https://github.com/capawesome-team/capacitor-firebase/blob/main/packages/authentication/docs/firebase-js-sdk.md)) or you could store the information on sign-in in local storage as workaround.

**Reply by Danielmn94** [reply]
Thanks for your replies. 
That was a good idea to use the local storage for a workaround, i'll try that. 
Thank you.
